docs/dg/dev/acp/integrate-acp-payment-apps-with-spryker-oms-configuration.md

```diff
@@ -33,31 +39,102 @@ The main OMS file includes `<subprocesses>` – similar to libraries or building
+Transitions between states in payment apps can be handled in two ways:  
 
-The list of payment event messages is predefined, and they are common for all payment methods from the ACP App Catalog:
-- PaymentAuthorized
-- PaymentAuthorizationFailed
-- PaymentCanceled
-- PaymentCancellationFailed
-- PaymentCaptured
-- PaymentCaptureFailed
-- PaymentRefunded
-- PaymentRefundFailed
+- Event-based: Trigger events based on Payment App messages
+- Conditional: Use conditional transitions based on Payment App messages and condition plugins
+
+Each approach requires a different setup:  
+
+- Event-based: Add `Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin` to `MessageBrokerDependencyProvider`
+- Conditional: Add `Spryker\Zed\PaymentApp\Communication\Plugin\MessageBroker\PaymentAppOperationsMessageHandlerPlugin` to `MessageBrokerDependencyProvider` along with the necessary condition plugins
+
+The following sections provide detailed instructions for configuring each option.  
+
+
+
+
+
+
+
+#### Transition based on event triggers  
+
+The `Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin` contains message handlers that trigger events based on messages received from the Payment App.  
+
+State transitions occur automatically through asynchronous ACP messages, processed by the `spryker/message-broker` module. The following sub-processes have auto-transitions:
+
+* `PaymentAuthorization`
+* `PaymentCapture`
+* `PaymentRefund`
+* `PaymentCancel`
+
+The MessageBroker worker runs in the background as a cron job, checking for new messages and triggering OMS events based on the configuration in `Spryker\Zed\Payment\PaymentConfig::getSupportedOrderPaymentEventTransfersList()`. You can modify this configuration to fit your project’s needs.  
+
+{% info_block infobox %}
+
+Use this approach when closely following the default `ForeignPaymentStateMachine01`, particularly if your OMS does not contain slow-running commands that could cause transition failures.  
+
+The payment status on the app side can change rapidly, which may cause issues if your OMS is not in the correct state for the transition. For example, if a command between `PaymentAuthorized` and `PaymentCapturePending` takes too long to process and a message arrives attempting to transition from `PaymentCapturePending` to `PaymentCaptured`, the OMS may not be able to complete the transition because it is not yet in the required state.  
+
+{% endinfo_block %}
+
+#### Conditional transitions
+
+The `\Spryker\Zed\PaymentApp\Communication\Plugin\MessageBroker\PaymentAppOperationsMessageHandlerPlugin` works differently then the above. When a message is received from the Payment App, the message handler will persist a status of the payment that is given on the App side. Through conditions (list down below) the OMS will check if the payment is in a state that allows the transition to the next state.
```

Comment:
Update the first sentence to something similar to https://github.com/spryker/spryker-docs/pull/3053/files#diff-3743ce689a6db4689fb7f3d245fcf93f585e973b27238fc3ed76cb3347989564R80

Reply:
Updated